README.md

```diff
@@ -206,20 +207,24 @@ export default container => {
       };
 
       // Don't fetch data for initial route, server has already done the work:
-      if (window.INITIAL_STATE) {
-        // Delete global data so that subsequent data fetches can occur:
-        delete window.INITIAL_STATE;
-      } else {
-        // Fetch mandatory data dependencies for 2nd route change onwards:
-        trigger('fetch', components, locals);
-      }
-
-      // Fetch deferred, client-only data dependencies
-      trigger('defer', components, locals)
-        // Finally, trigger 'done' lifecycle hooks:
-        .then(() => trigger('done', components, locals));
+      const invoke = window.INITIAL_STATE ? Promise.resolve.bind(Promise) : trigger;
+      
+      invoke('fetch', components, locals) // Fetch mandatory data dependencies for 2nd route change onwards
+      .then(() => trigger('defer', components, locals)) // Fetch deferred, client-only data dependencies
+      .then(() => trigger('done', components, locals)) // Finally, trigger 'done' lifecycle hooks
+      .then(callback);
```

Comment:
is all this working for you? because from my point of way this will defer transition until all is resolved when hooked to `history.listenBefore`